tests/core/running/test_running_with_logging.py

```diff
@@ -1,6 +1,33 @@
+import pytest
+import threading
+import sys
+
 from geth import DevGethProcess
 from geth.mixins import LoggingMixin
 
+_ERRORS = []
+
+
+@pytest.fixture(autouse=True)
+def fail_from_errors_on_other_threads():
```

Comment:
Ha this was trickier than I thought it would be. The error was happening on another thread so that is why it was not failing the test originally.

Now, it should detect errors from other threads in this pytest fixture and raise if it finds them.
Might be worth putting this in a more global `contest` but I am kind of scared to do that!

Reply:
If you copy + paste this code into the master branch, it fails

Reply:
This test doesn't actually fail for me on master and never reaches the manual trigger to fail. I think its the case that since StopIteration exception message is just logged out by the mixin instead of actually raising.

That said, I confirmed that we see the StopIteration logging output on master and that the code change removes that, so I believe this PR is worth merging without the test.

Double-checking: you actually see failed tests on your machine? If so, what version of python and pytest are you using?

Reply:
Yes the test fails for me, I am using python 3.9.7 and geth 1.10.1.

I am curious if the test fails for you when you try it this way instead:

```python
import pytest
import threading
import sys

from geth import DevGethProcess
from geth.mixins import LoggingMixin

_ERRORS = []


@pytest.fixture(autouse=True)
def fail_from_errors_on_other_threads():
    """
    Causes errors when `LoggingMixin` is improperly implemented.
    Useful for preventing false-positives in logging-based tests.
    """
    global _ERRORS
    _ERRORS = []

    def pytest_excepthook(*args, **kwargs):
        _ERRORS.extend(args)

    sys.excepthook = pytest_excepthook
    threading.excepthook = pytest_excepthook

    yield

    sys.excepthook = sys.__excepthook__
    if _ERRORS:
        caught_errors_str = ', '.join([str(err) for err in _ERRORS])
        pytest.fail(f'Caught exceptions from other threads:\n{caught_errors_str}')


class WithLogging(LoggingMixin, DevGethProcess):
    pass


def test_with_logging(base_dir):
    geth = WithLogging('testing', base_dir=base_dir)

    geth.start()

    assert geth.is_running
    assert geth.is_alive

    geth.stop()

```

Reply:
Ah, interesting. I happened to be using 3.7.12 locally, but confirmed with later versions that tests do properly fail during teardown. To answer your last question, the updated test using `sys` doesn't appear to make a difference. I'm not terribly concerned about finding a way to make older versions of python fail this test, so I'm happy to put the LGTM stamp on it. Thanks!